core/arxiv/submission/process/checks/tex_produced/__init__.py

```diff
@@ -0,0 +1,354 @@
+"""arXiv TeX produced routines.
+
+This package checks a submission file object to determine whether it was generated
+from TeX. arXiv requires the TeX source, when available, to make long-term maintenance
+of papers easier. Single-file submission determined to be generated by TeX are rejected.
+
+The tex-produced check current works on both streams and files for PDFs and just files
+for Postscript files. The original code only worked with files. In NG we pass around streams
+of file content which would require writing out in order to user the file-based check.
+
+We now convert the non-seekable stream to an in-memory object and use the PyPDF2 package
```

Comment:
I was trying to avoid writing the file to disk just for this test. I believe the submission size limits keep memory use to a manageable size. I'm open to other solutions. Maybe create a FileManager-style app layer that keeps the file in a RAM disk to avoid AWS io charges. ^:(

Reply:
:+1:  If it becomes a problem we'll figure out a solution.